rasa/core/policies/policy.py

```diff
@@ -372,15 +372,16 @@ def predict_action_probabilities(
         self,
         tracker: DialogueStateTracker,
         domain: Domain,
-        precomputations: Optional[MessageContainerForCoreFeaturization] = None,
+        rule_only_data: Optional[Dict[Text, Any]] = None,
```

Comment:
This is changing the common policy interface here so that it matches what is needed in `Memoization` and `RulePolicy` - but  `TEDPolicy` needs the precomputations and we need to be able to find that parameter via inspection I think. So I think we should add both (or remove this part of the common policy interface - in which case we would have to remove the related tests in `test_policy.py` which assume this common interface).

Reply:
I added this to the interface because _every_ policy requires `rule_only_data` while only policies with end-to-end support require `precomputations`. The graph is flexible enough to provide data only to the components which need it and in my opinion the `kwargs` cover for this case.

> we need to be able to find that parameter via inspection I think

In this case we can just inspect `TEDPolicy` which has this param in its signature, can't we?


> we would have to remove the related tests in test_policy.py

done 👍🏻

Reply:
>The graph is flexible enough to provide data only to the components which need it and in my opinion the kwargs cover for this case.

Should we then adapt TEDPolicy etc to implement the exact same interface and pass/retrieve the precomputations as/from kwargs?

Reply:
In my opinion it's fine to keep it in the signature and also have it explicit. Also doesn't violate the interface contract as it has a default